libs/go/aealite/protocols/message.go

```diff
@@ -0,0 +1,178 @@
+/* -*- coding: utf-8 -*-
+* ------------------------------------------------------------------------------
+*
+*   Copyright 2018-2019 Fetch.AI Limited
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       http://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*
+* ------------------------------------------------------------------------------
+ */
+
+package protocols
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type MessageId int
+type Address string
+type Performative string
+
+type ProtocolMessageInterface interface {
+	Sender() Address
+	SetSender(Address) error
+	To() Address
+	SetTo(Address) error
+	MessageId() MessageId
+	DialogueReference() DialogueReference
+	Target() MessageId
+	Performative() Performative
+	Body() map[string]interface{}
+	HasSender() bool
+	HasTo() bool
+	GetField(name string) interface{}
+	//ValidPerformatives() []Performative TODO temporarily removed
+}
+
+type DialogueMessageWrapper struct {
```

Comment:
Because `DialogueMessage` has a Content field which is of type `byte[]`; that would mean to access an attribute it is required to parse the content of `byte[]` and access the required field. Also, that has to be done manually, as the user is interacting directly with `DialogueMessage`.

With this class, instead, we would do it only once, and access the fields saving the cost of decoding the `Content` field each time.